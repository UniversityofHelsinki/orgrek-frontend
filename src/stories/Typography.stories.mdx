import { Meta, Canvas } from '@storybook/addon-docs';
import { Stack, Typography } from '@mui/material';

<Meta title="Typography" />

# Typography

Fonts and styles for MUI [Typography](https://mui.com/material-ui/react-typography/) component adapted from
[HY Design System](https://wiki.helsinki.fi/pages/viewpage.action?pageId=334495865) specs.

Note that `Typography` supports defining the semantic component separately from the style so that the heading
hierarchy can be organized correctly (see [Accessibility](https://mui.com/material-ui/react-typography/#accessibility)).

<Canvas>
  <Stack gap={2}>
    <Typography variant="h1">
      h1. Lorem ipsum dolor sit amet,
      <br />
      consectetur adipiscing elit.
    </Typography>
    <Typography variant="h2">
      h2. Lorem ipsum dolor sit amet,
      <br />
      consectetur adipiscing elit.
    </Typography>
    <Typography variant="h3">
      h3. Lorem ipsum dolor sit amet,
      <br />
      consectetur adipiscing elit.
    </Typography>
    <Typography variant="h4">
      h4. Lorem ipsum dolor sit amet,
      <br />
      consectetur adipiscing elit.
    </Typography>
    <Typography variant="h5">
      h5. Lorem ipsum dolor sit amet,
      <br />
      consectetur adipiscing elit.
    </Typography>
    <Typography variant="h6">
      h6. Lorem ipsum dolor sit amet,
      <br />
      consectetur adipiscing elit.
    </Typography>
    <Typography variant="ingress">
      ingress. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras
      dapibus vulputate diam eu pretium. Mauris elit orci, ultricies id
      fermentum vel, porta et eros. Vestibulum condimentum lectus in convallis
      feugiat. Sed vulputate fringilla felis.
    </Typography>
    <Typography variant="body1" component="div">
      <p>
        body1. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras
        dapibus vulputate diam eu pretium. Mauris elit orci, ultricies id
        fermentum vel, porta et eros. Vestibulum condimentum lectus in convallis
        feugiat. Sed vulputate fringilla felis. Aliquam ut arcu et dui feugiat
        scelerisque eu quis diam. Mauris placerat congue dui sit amet blandit.
        Phasellus condimentum libero vel velit auctor, sit amet tincidunt velit
        varius.
      </p>
      <p>
        Mauris lacinia porta faucibus. Fusce eu est ac eros vulputate mollis in
        ac felis. Aenean commodo scelerisque mi sed imperdiet. Donec at
        hendrerit nisi, eget vestibulum nisi. Sed sit amet magna luctus,
        facilisis erat quis, sagittis ligula. Aenean dignissim velit quis leo
        consequat ultricies. Proin quis pretium justo. Vestibulum at eros nisl.
        Fusce lobortis erat ante, eu cursus sapien molestie at. Pellentesque
        placerat ante diam, et euismod lacus dictum vel. Phasellus vitae
        sollicitudin mi.
      </p>
      <p>
        In pulvinar eleifend convallis. Suspendisse elit erat, venenatis eget
        ullamcorper ut, laoreet iaculis nisl. Sed porta, felis id rhoncus
        aliquet, quam ipsum pellentesque metus, in sodales quam nunc vitae
        risus.
      </p>
    </Typography>
    <Typography variant="button">Button text</Typography>
    <Typography variant="navigationLink">www.navigationLink.fi</Typography>
    <Typography variant="caption">Caption text</Typography>
    <Typography variant="label">Label text</Typography>
    <Typography variant="accordionTitle">Accordion title</Typography>
    <Typography variant="tableHead">Table head</Typography>
    <Typography variant="tableBody">Table body</Typography>
  </Stack>
</Canvas>
