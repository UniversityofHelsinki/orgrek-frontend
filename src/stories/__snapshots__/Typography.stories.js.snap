// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Primary smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "H1. LOREM IPSUM DOLOR SIT AMET, CONSECTETUR ADIPISCING ELIT.",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "level": 2,
      "name": "h2. Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "level": 3,
      "name": "h3. Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "level": 4,
      "name": "h4. Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "level": 5,
      "name": "h5. Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "level": 6,
      "name": "h6. Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "ingress. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras dapibus vulputate diam eu pretium. Mauris elit orci, ultricies id fermentum vel, porta et eros. Vestibulum condimentum lectus in convallis feugiat. Sed vulputate fringilla felis.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "body1. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras dapibus vulputate diam eu pretium. Mauris elit orci, ultricies id fermentum vel, porta et eros. Vestibulum condimentum lectus in convallis feugiat. Sed vulputate fringilla felis. Aliquam ut arcu et dui feugiat scelerisque eu quis diam. Mauris placerat congue dui sit amet blandit. Phasellus condimentum libero vel velit auctor, sit amet tincidunt velit varius.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Mauris lacinia porta faucibus. Fusce eu est ac eros vulputate mollis in ac felis. Aenean commodo scelerisque mi sed imperdiet. Donec at hendrerit nisi, eget vestibulum nisi. Sed sit amet magna luctus, facilisis erat quis, sagittis ligula. Aenean dignissim velit quis leo consequat ultricies. Proin quis pretium justo. Vestibulum at eros nisl. Fusce lobortis erat ante, eu cursus sapien molestie at. Pellentesque placerat ante diam, et euismod lacus dictum vel. Phasellus vitae sollicitudin mi.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "In pulvinar eleifend convallis. Suspendisse elit erat, venenatis eget ullamcorper ut, laoreet iaculis nisl. Sed porta, felis id rhoncus aliquet, quam ipsum pellentesque metus, in sodales quam nunc vitae risus.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Button text",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Navigation link",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Caption text",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "LABEL TEXT",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Accordion title",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Table head",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Table body",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "typography--primary",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
